internal.nix

```diff
@@ -331,6 +331,9 @@ rec {
 
               ${preInstallLinkCommands}
 
+              jq -r '. | select(.bin != null) | .bin | values[]' package.json | while read binTarget; do
```

Comment:
`bin` field could also have string as the value if there is a single executable[1]
```
{
  "name": "my-program",
  "version": "1.2.5",
  "bin": "./path/to/program"
}
```
Currently, this `jq` expression doesn't seem to handle that case. I used something like `'select(has("bin")) | .bin | if type == "string" then . else .[] end'` to handle that case. But this is the first time I wrote `jq` expression, so it might be far from the optimal.

[1]: https://docs.npmjs.com/cli/v7/configuring-npm/package-json#bin

Reply:
thanks!

Reply:
I am not really into maintaining jq expression.. I've done it once or twice but generally that isn't something that i want to maintain beyond a certain level of complexity.

How much effort would it be to rewrite this in a short nodejs script (without dependencies)? The conversaion so far makes me fear about any additional special cases that we will have to add in the future.

Reply:
jq is a standard tool for json manipulation, node feels like overkill for this

> The conversaion so far makes me fear about any additional special cases that we will have to add in the future.

heh, not really ... there are three cases

the `bin` field ...

1. does not exist
2. is a string
3. is an object

all of these are handled

https://docs.npmjs.com/cli/v7/configuring-npm/package-json#bin